### PR TITLE
Prefer `Action<T>` to `T.() -> Unit` for Kotlin DSL accessors

### PR DESCRIPTION
**Pull request by bamboo**
To workaround a member resolution bias in the Kotlin compiler that causes extension members accepting a Kotlin function type (e.g. `T.() -> Unit`) to be preferred over instance members accepting `Action<T>`.

To be rebased on top of master and merged after #14863 and #14867 are merged.

### CODE REVIEW

**Comment by bamboo** [comment]
@bot-gradle test this

**Comment by bamboo** [comment]
@bot-gradle test this

**Comment by eskatos** [comment]
@bot-gradle test this

File: subprojects/kotlin-dsl/src/test/kotlin/org/gradle/kotlin/dsl/accessors/ProjectAccessorsClassPathTest.kt [hunk 1]
```diff
@@ -386,6 +386,13 @@ class ProjectAccessorsClassPathTest : AbstractDslTest() {
                         }
                     }
                 }
+
+                fun Project.canUseContainerElementAccessors() {
```
Comment (eskatos) [inline subprojects/kotlin-dsl/src/test/kotlin/org/gradle/kotlin/dsl/accessors/ProjectAccessorsClassPathTest.kt:390]: 👍

**Review by eskatos — APPROVED** [review]
Confirmed this fixes the issue reproduced in https://github.com/eskatos/kt-1.4-zac-issue
The two dependent PRs are now merged into `master`

**Comment by eskatos** [comment]
@bot-gradle test this

**Comment by eskatos** [comment]
@bot-gradle test this

**Comment by eskatos** [comment]
Published `6.8-branch-bamboo_master_kotlin_dsl_prefer_action_over_kotlin_function_types_i-20201015081032+0000` so this can be tried easily

**Comment by bamboo** [comment]
Thanks, @eskatos!

**Pull request merged**